feature-libs/quote/components/summary/actions/quote-summary-actions.component.html

```diff
@@ -2,6 +2,17 @@
   <ng-container *ngIf="cartDetails$ | async as cartDetails">
     <ng-container *ngIf="areButtonsRendered(quoteDetails.allowedActions)">
       <section [ngClass]="isFixedPosition ? 'fixed' : 'sticky'">
+        <button
+          type="button"
+          id="downloadBtn"
+          class="btn btn-primary"
+          *ngIf="isAttachmentPresent(quoteDetails.sapAttachments)"
+          (click)="
+            onDownloadAttachment(quoteDetails.code, quoteDetails.sapAttachments)
+          "
+        >
+          {{ 'quote.summary.actions.DOWNLOAD' | cxTranslate }}
```

Comment:
MEDIUM:
can we try this suggestion to maintain uniformity and load the action type as DOWNLOAD?

```suggestion
          {{ 'quote.summary.actions.' + action.type | cxTranslate }}
```

Reply:
We don't get `DOWNLOAD` as one of the values of `allowedActions.type` property from the API response, so won't be able to code as per your suggestion.

Reply:
But are we allowed to download attachments for every quote state? (As all other actions depend on the state)

Reply:
No we are not allowed to download attachments for every quote state. 
Only when a vendor generates and sends back a proposal document from CPQ, the status for the quote is displayed as **Vendor Quote** in Spartacus UI, and then we are allowed to download it as an attachment.

Reply:
And where is the check for this specific status? Currently the button appears if attachments are present (not depending on any status)

Reply:
Checking for the presence of an attachment is a more suitable condition for displaying the button IMO, as I am unaware if there are any other states besides **Vendor Quote** where the attachment might be present.

Reply:
Regarding two consecutive primary buttons I have to check with our UI designer, but I strongly believe that she will not be happy with it.

Reply:
Suppose a quote is in status VENDOR_QUOTE and carries a proposal. Then let the buyer edit the quote again, change a qty. In this case, no proposal download should be possible, would you agree?
How do we achieve that? Or will the backend in this case send no proposal?

Reply:
> Suppose a quote is in status VENDOR_QUOTE and carries a proposal. Then let the buyer edit the quote again, change a qty. In this case, no proposal download should be possible, would you agree? How do we achieve that? Or will the backend in this case send no proposal?

@suprishi Could you please answer this one?

Reply:
@ChristophHi apologies for missing this conversation.
In the scenario you described, the quote would change its status to **Submitted** again, and yes, the user would not be able to download the attachment anymore.
The backend would send an empty array in the `sapAttachments` key in the response, which is what I am using to determine if the download button should be displayed.